libzr/src/math/shifts.c

```
@@ -0,0 +1,30 @@
+    type shl_##name(type value, uint32_t shift) \
+    {                                           \
+        return value << shift;                  \
+    }
+
+#define BUILD_SHR_FUNC(name, type)              \
+    type shr_##name(type value, uint32_t shift) \
+    {                                           \
+        return value >> shift;                  \
+    }
+
+BUILD_SHL_FUNC(i8, uint8_t)
```

Cppcheck (2.19.0)

[style] The function 'shl_i8' is never used.

(unusedFunction)